### ISSUE TITLE
Inflector fails with a StackOverFlow Exception with circular ref definitions

### ISSUE THREAD

**Issue by gracekarina** [swagger-api/swagger-inflector#248]

**Comment by shannon7wallace** [comment]
Hey Grace, how is this coming along? We're hoping to pull this into SwaggerHub whenever it's ready.

**Comment by gracekarina** [comment]
Hi, I think today will be ready.

**Comment by gracekarina** [comment]
It's merged! @joeljons @shannon7wallace

**Comment by m-aamir-ashfaq** [comment]
I am still getting the issue when circular reference is in response.
```
swagger: '2.0'
tags:
- name: candidate
# - http
paths:
  /candidates:
    get:
      operationId: getCandidate
      responses:
        '200':
          description: successful response
          schema:
            $ref: '#/definitions/Candidate'
  /sources:
    get:
      operationId: getSource
      responses:
        '200':
          description: successful response
          schema:
            $ref: '#/definitions/Source'
definitions:
  Candidate:
    type: object
    required:
    - firstName
    - lastName
    properties:
      id:
        type: integer
        format: int64
      firstName:
        type: string
        example: Jean
      lastName:
        type: string
        example: Dupont
      source:
        $ref: '#/definitions/Source'
  Source:
    type: object
    required:
    - name
    properties:
      id:
        type: integer
        format: int64
      name:
        type: string
        example: CDR
      candidates:
        $ref: '#/definitions/Candidate'
```

and tried following code:

      ```
  Swagger swagger = new SwaggerParser().read("./src/test/swagger/circuler-refs-SPLAT-56-2.yaml");
        ResolverUtil resolverUtil = new ResolverUtil();
        resolverUtil.resolveFully(swagger);
        Response response = swagger.getPaths().get("/candidates").getOperationMap().get(HttpMethod.GET).getResponses().get("200");
        Example example = ExampleBuilder.fromModel("", response.getResponseSchema(), swagger.getDefinitions(), new HashSet<String>());
        assertNotNull(example);
```

**Comment by gracekarina** [comment]
Hi this was closed by #266 thanks @m-aamir-ashfaq
